src/fping.c

```diff
@@ -2486,8 +2496,11 @@ int wait_for_reply(int64_t wait_time)
         if (verbose_flag || alive_flag) {
             printf("%s", h->host);
 
-            if (verbose_flag)
+            if (verbose_flag) {
                 printf(" is alive");
+                if(print_tos_flag && ip_header_tos != -1)
+                    printf(" (TOS %d)", ip_header_tos);
+            }
```

Comment:
I think this is OK. Alternatively, with `print_tos_flag && ip_header_tos == -1`, you could print ` (TOS unknown)`. I am not sure what is better, your variant is OK as it is.

Reply:
I have changed it